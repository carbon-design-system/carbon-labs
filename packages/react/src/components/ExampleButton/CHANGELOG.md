# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2025-01-30)


### Features

* script to generate base component in react + web-components ([#354](https://github.com/carbon-design-system/carbon-labs/issues/354)) ([3c24d37](https://github.com/carbon-design-system/carbon-labs/commit/3c24d375f2b8773f7c4dd704197ae6267d919685))
