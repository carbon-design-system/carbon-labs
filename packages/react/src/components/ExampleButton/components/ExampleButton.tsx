/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import { Button, ButtonProps } from '@carbon/react';
import { usePrefix } from '@carbon-labs/utilities/es/index.js';

/** Primary UI component for user interaction */

interface ExampleButtonProps extends ButtonProps<any> {
  children?: React.ReactNode;
}

export const ExampleButton = ({
  children = 'Button',
  ...rest
}: ExampleButtonProps) => {
  const prefix = usePrefix();

  return (
    <div className={`${prefix}--example-button__container`}>
      <Button {...rest}>{children}</Button>
    </div>
  );
};
