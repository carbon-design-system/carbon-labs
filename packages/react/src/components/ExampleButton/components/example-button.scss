/**
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

@use '@carbon/styles/scss/theme' as *;
@use '@carbon/styles/scss/colors' as *;
@use '@carbon/styles/scss/spacing' as *;

$prefix: 'clabs' !default;

.#{$prefix}--example-button__container {
  padding: $spacing-01;
  border: 1px solid $border-subtle-01;
  inline-size: fit-content;
}
