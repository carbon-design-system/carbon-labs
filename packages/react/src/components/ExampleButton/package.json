{
  "name": "@carbon-labs/react-example-button",
  "version": "0.0.0",
  "publishConfig": {
    "access": "public",
    "provenance": true
  },
  "type": "module",
  "description": "Carbon Labs - Example Button",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/carbon-design-system/carbon-labs",
    "directory": "src/components/ExampleButton"
  },
  "main": "./src/index.js",
  "module": "./src/index.js",
  "files": [
    "es",
    "lib",
    "scss"
  ],
  "scripts": {
    "build": "node ../../../tasks/build.js",
    "clean": "rm -rf {es,lib,scss}"
  },
  "devDependencies": {
    "@carbon-labs/utilities": "canary"
  }
}
