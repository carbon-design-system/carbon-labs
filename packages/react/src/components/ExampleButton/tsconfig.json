{
  "extends": "typescript-config-carbon/tsconfig.base.json",
  "compilerOptions": {
    // TODO: Turn back on once stricter typings for internal utitlies are complete
    "noImplicitAny": false
  },
  "exclude": ["**/__stories__/*.stories.js", "**/__tests__/*", "**/*d.ts"]
}
