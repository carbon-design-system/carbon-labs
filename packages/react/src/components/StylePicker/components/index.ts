/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { StylePickerColor } from './style-picker-color';
export { StylePickerIcon } from './style-picker-icon';
export { StylePickerGroup } from './style-picker-group';
export { StylePickerSection } from './style-picker-section';
export { StylePickerSections } from './style-picker-sections';
export { StylePickerOption } from './style-picker-option';
export { StylePicker } from './style-picker';
