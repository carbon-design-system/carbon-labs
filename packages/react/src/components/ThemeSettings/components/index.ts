/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { ThemeSettings } from './ThemeSettings';
export { ThemeSwitcher } from './ThemeSwitcher';
export { ThemeSetDropdown } from './ThemeSetDropdown';
export { ThemeMenuComplement } from './ThemeMenuComplement';
export { themeSets } from './theme-settings-consts';
