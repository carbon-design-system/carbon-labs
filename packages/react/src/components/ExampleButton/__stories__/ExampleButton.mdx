import { ArgTypes, Canvas, Meta } from '@storybook/blocks';
import * as ExampleButtonStories from './ExampleButton.stories';

<Meta isTemplate />

# ExampleButton

{/* <!-- START doctoc generated TOC please keep comment here to allow auto update --> */}
{/* <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE --> */}

> ðŸ’¡ Check our
> [Stackblitz](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/example-button)
> example implementation.

[![Edit carbon-labs](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/example-button)

## Table of Contents

- [Overview](#overview)
- [Getting started](#getting-started)

{/* <!-- END doctoc generated TOC please keep comment here to allow auto update --> */}

## Overview

This is an example button component.

<Canvas of={ExampleButtonStories.Default} />

## Getting started

Here's a quick example to get you started.

```bash
yarn add @carbon/react
yarn add @carbon-labs/react-example-button
```

### JS (via import)

```javascript
import { ExampleButton } from '@carbon-labs/react-example-button/es/index';

function App() {
  return <ExampleButton />;
}

export default App;
```

### SCSS

In your styles file import

```
@use '@carbon/react';
@use '@carbon-labs/react-example-button/scss/example-button';
```
