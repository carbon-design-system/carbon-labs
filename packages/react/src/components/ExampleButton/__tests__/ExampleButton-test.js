/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import '@testing-library/jest-dom';

import { ExampleButton } from '../components/ExampleButton';
jest.mock('./button.scss', () => ({}));
describe('ExampleButton', () => {
  describe('renders as expected - Component API', () => {
    it('should match snapshot', () => {
      const { container } = render(<ExampleButton />);
      expect(container).toMatchSnapshot();
    });

    it('renders a button element', () => {
      render(<ExampleButton />);
      expect(screen.getByRole('button')).toBeInTheDocument();
    });

    it('renders with default text "Button"', () => {
      render(<ExampleButton />);
      expect(screen.getByRole('button')).toHaveTextContent(/^Button$/);
    });

    it('supports a custom class name', () => {
      const { container } = render(<ExampleButton className="test" />);
      expect(container.firstChild.firstChild).toHaveClass('test');
    });

    it('supports additional props', () => {
      render(<ExampleButton data-testid="test-button" />);
      expect(screen.getByRole('button')).toHaveAttribute(
        'data-testid',
        'test-button'
      );
    });

    it('supports disabled state', () => {
      render(<ExampleButton disabled />);
      expect(screen.getByRole('button')).toBeDisabled();
    });
  });

  describe('behaves as expected', () => {
    it('calls onClick handler when clicked', async () => {
      const onClick = jest.fn();
      render(<ExampleButton onClick={onClick} />);

      expect(onClick).toHaveBeenCalledTimes(0);
      await userEvent.click(screen.getByRole('button'));
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
