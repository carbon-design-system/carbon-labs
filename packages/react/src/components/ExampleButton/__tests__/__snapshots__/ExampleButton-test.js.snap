// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleButton renders as expected - Component API should match snapshot 1`] = `
<div>
  <div
    class="clabs--example-button__container"
  >
    <button
      class="cds--btn cds--btn--primary"
      type="button"
    >
      Button
    </button>
  </div>
</div>
`;
